cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/web/HomeController.java

```diff
@@ -35,6 +57,27 @@ public String home(Model model) {
         //model.addAttribute("journals", pubDao.getJournals());
         model.addAttribute("journalCount", pubDao.getJournalCount());
         //model.addAttribute("textCountriesCount", pubDao.getCountriesInTextCount());
+        
+		ArrayList<String> countriesList = new ArrayList<String>();
+        try (CloseableIterator<CountriesImpl> countries = mongoTemplate.stream(new Query().noCursorTimeout(), CountriesImpl.class)) {
+            while(countries.hasNext()) {
+                CountriesImpl country = countries.next();
+                if(country.getProperties().getSelectedWikipediaCount() > 0) {
+					ObjectMapper objectMapper = new ObjectMapper();
+					objectMapper.disableDefaultTyping();
+					SimpleBeanPropertyFilter theFilter = SimpleBeanPropertyFilter
+					          .serializeAllExcept("id", "@type");
+					      FilterProvider filters = new SimpleFilterProvider()
+					          .addFilter("myFilter", theFilter);
+					String str = objectMapper.writer(filters).writeValueAsString(country);
+					countriesList.add(str);
+                }
+            }
+        }

```

Comment:
this is business logic and should not be done in a controller

Reply:
Where should the serialization be done then?

Reply:
Put it in its own class (e.g. a countryManager)